clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -1344,6 +1377,7 @@ public void run() {
 
       while (isRunning()) {
         Store store = storeRepository.getStoreOrThrow(storeName);
+        updateIngestionRoleIfStoreChanged(store);
```

Comment:
Why do this as the first step? Is there any particular reason?

Reply:
No particular reason, but just want to get changed version role or workloadType information asap to update internal state. As changed information inside SIT could affect following subscriptions executed in this loop.